WordPress/gutenberg
# Fix visually hidden classnames

## Description
While reviewing #20607, I noticed `VisuallyHidden` doesn't support a `className` prop. If `className` is specified, the existing `components-visually-hidden` class is overwritten, making the component no longer hidden.

There was a use case where spreading props onto `VisuallyHidden` seemed like a reasonable option (https://github.com/WordPress/gutenberg/pull/20607#discussion_r388148496), so I thought this would be an opportunity to change the component to better support class names.

I think there are two options:
1. Class names received through the `className` prop are appended.
2. Class names received through the `className` prop are ignored.

I went with the first option for consistency with other components, but I think the second also has merit—it'd mean other styles couldn't break `VisuallyHidden`. Happy to change the approach based on feedback.

## How has this been tested?
Added snapshot tests.

For manual testing:
1. Find a VisuallyHidden component
2. Add a className prop
3. Observe that the class name is appended to the components class name.

## Types of changes

Bug fix (non-breaking change which fixes an issue)



## Checklist:
